From 4881da8bf9d7e76db165cdfbca65ee49b5104eb3 Mon Sep 17 00:00:00 2001
From: Nate Myren <ntmyren@google.com>
Date: Thu, 10 Nov 2022 14:09:34 -0800
Subject: [PATCH] RESTRICT AUTOMERGE Use static token for myAttributionSource
 in ServiceUtilities

Bug: 258672042
Test: manual
Change-Id: I44243bbbd68db9ee8b5c4ba02a061e14bd300d20
(cherry picked from commit ca6e2770753c5d61185552140abd613367b79365)
Merged-In: I44243bbbd68db9ee8b5c4ba02a061e14bd300d20
---
 media/utils/ServiceUtilities.cpp | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/media/utils/ServiceUtilities.cpp b/media/utils/ServiceUtilities.cpp
index 07f45298ab..83b84e392c 100644
--- a/media/utils/ServiceUtilities.cpp
+++ b/media/utils/ServiceUtilities.cpp
@@ -103,11 +103,10 @@ std::optional<AttributionSourceState> resolveAttributionSource(
     AttributionSourceState myAttributionSource;
     myAttributionSource.uid = VALUE_OR_FATAL(android::legacy2aidl_uid_t_int32_t(getuid()));
     myAttributionSource.pid = VALUE_OR_FATAL(android::legacy2aidl_pid_t_int32_t(getpid()));
-    if (callerAttributionSource.token != nullptr) {
-        myAttributionSource.token = callerAttributionSource.token;
-    } else {
-        myAttributionSource.token = sp<BBinder>::make();
-    }
+    // Create a static token for audioserver requests, which identifies the
+    // audioserver to the app ops system
+    static sp<BBinder> appOpsToken = sp<BBinder>::make();
+    myAttributionSource.token = appOpsToken;
     myAttributionSource.next.push_back(nextAttributionSource);
 
     return std::optional<AttributionSourceState>{myAttributionSource};
-- 
2.39.0.rc1.256.g54fd8350bd-goog

